git/indexpack.go

```diff
@@ -131,13 +131,25 @@ func (idx PackfileIndexV2) getObjectAtOffset(r io.ReaderAt, offset int64, metaOn
 	metareader := io.NewSectionReader(r, offset, 4096)
 	t, sz, ref, refoffset, rawheader := p.ReadHeaderSize(metareader)
 	var rawdata []byte
-	// sz is the uncompressed size, so the total size should be less than
-	// sz for the compressed data. It might theoretically be a little more,
-	// but we're very generous here since this doesn't allocate anything but
-	// just determines how much data the SectionReader will read before
-	// returning an EOF.
+	// sz is the uncompressed size, so the total size should usually be
+	// less than sz for the compressed data. It might theoretically be a
+	// little more, but we're generous here since this doesn't allocate
+	// anything but just determines how much data the SectionReader will
+	// read before returning an EOF.
+	//
+	// There is still overhead if the underlying ReaderAt reads more data
+	// than it needs to then discards it, so we assume that it won't
```

Comment:
```suggestion
	// than it needs to and then discards it, so we assume that it won't
```

can't figure out how to make this edit through github's ui without suggesting changes..